src/mrsimulator/methods/base.py

```diff
@@ -157,12 +157,16 @@ def check_method_compatibility(cls, py_dict):
         default_method = cls.update(**py_dict)
         default_spectral_dimensions = default_method["spectral_dimensions"]
         for i, item in enumerate(py_dict["spectral_dimensions"]):
-            if item["events"] == [{}]:
+
+            # If no methods in SpectralDimension, set to default events
+            if item["events"] == [{}] or item["events"] == []:
                 item["events"] = default_spectral_dimensions[i]["events"]
 
             elif item["events"] != default_spectral_dimensions[i]["events"]:
                 raise ImmutableEventError(cls.__name__)
 
+            py_dict["spectral_dimensions"][i] = item
```

Comment:
What is the purpose of this statement? `item` is the ith spectral dimension reference from `py_dict`, so this statement is `item = item`!

Reply:
line has been removed